### DISCUSSION TITLE
neb-mouse-rna-xcr-umi-nebnext QC ALERT

### DISCUSSION THREAD

**Discussion by Victor-K27** [milaboratory/mixcr#1768] [Q&A] [answered]
Hi all,

I am running the `neb-mouse-rna-xcr-umi-nebnext  analyze` preset of the MiXCR package. 

I am getting a low number of successfully aligned reads, and a high number of off-target reads. 
I am also getting a lot of alignments and tags that do not cover the VDJRegion, and high artificial UMI diversity. 
I believe that this indicates weak enrichment for TCR sequences?

Regarding the trimming, I am getting an average of 5-20 nucleotides, which is hopefully ok?

I am attaching bellow a copy of the qc reports. I have been going through them, but some comments from someone more experienced would be very appreciated!

I am also wondering, is it normal to have that high ratio of TRB to TRA reads? In this sample for example, TRA reads represent 10% of the repertoire with TRB reads at 88%. 

I am wondering (as far as I was told this pipeline is new), if I am the only one experiencing issues?

[alignQc copy.pdf](https://github.com/user-attachments/files/16825428/alignQc.copy.pdf)
[Spleen_1.qc.txt](https://github.com/user-attachments/files/16825419/Spleen_1.qc.txt)
[Spleen_1.align.report.txt](https://github.com/user-attachments/files/16825420/Spleen_1.align.report.txt)
[Spleen_1.assemble.report.txt](https://github.com/user-attachments/files/16825421/Spleen_1.assemble.report.txt)
[Spleen_1.refine.report.txt](https://github.com/user-attachments/files/16825423/Spleen_1.refine.report.txt)

Thank you very much for your time! 
Victor

**Comment by mizraelson** [comment]
Hi Victor,
Generally speaking:
	•	A 30% alignment rate is quite low, and a high number of off-target reads suggests some primer misannealing.
	•	The refine report (UMI diversity) looks fine. It’s common for many UMIs to be filtered out, but the important thing is that more than 90% of reads were preserved.
	•	Trimming 5-20 nucleotides should be okay, and we still see that 90% of reads overlap. However, it’s strange that the full VDJ region is only covered in 30% of reads. It’s hard to say more without looking at the data.
	•	10% of TRA compared to TRB is quite low and might signify some wet lab issues.

If possible, you can share the raw FASTQ data (support@milaboratories.com) for this sample, and I will take a look and probably be able to provide more insights.

**Reply by mizraelson** [reply, marked as answer]
Hi Victor,
I looked into the non-aligned reads for one of the samples you shared. The majority of the off-target reads align to other genes (e.g., lacZ, Glyr1). Some reads do align to the TCR locus, but the alignment looks like this, for example:

```
CAAGGAGACCTTGGGTGGAGTCACATTTCTCAGATCCTCTGATGTTCCACATTCACTCCTACACTGAGATGTAAGAGAGCTGTGGTGGCTTGCATAGGAAAGCCGCAATCCTATTACCAAAAGCCTGGTCCCTGAGCCGAAGGTGTAGTCGGAGTTTGCACATCAGAATACAGATACTCGAATATGGACACGGAGGACATGCTTTGTCCGAAGAGAGACCTGGAAATTTACCTGAAAGATATCTTACCTACAACTGTGAGTCTGGTTCCTTTACCAAAGAAGACTTCTGTGTTTGCCCTGT
```

<img width="1120" alt="image" src="https://github.com/user-attachments/assets/4a56a442-e635-4054-af17-5d72d0cf2bb8">

This read covers two J genes and the DNA in between, which doesn’t resemble a regular rearranged TCR mRNA.

Or this:
```
CAAGGAGACCTTGGGTGGAGTCACATTTCTCAGATCCTCTGATGTTCCACATTCACTCCTACACTGAGATGTAAGAGAGCTGTGGTGGCTTGCATAGGAAAGCCGCAATCCTGTGGAAGATATTATTCACAGGTTGAACTCTTATTTTTTCAGTGCTCAGTGAACCTCCCTGATGAGTTTCAGAATTCTAGTTTACAACTAAATGTTCTATTCTTTAATATCCCGTAGATCAATGAATTTAGAGTATAAGATCGGAAGAGCACACGTCTAACTCCAGTCACATTACTCGATCTCGTATGCC
```

<img width="874" alt="image" src="https://github.com/user-attachments/assets/1a214274-a557-4415-ad98-9f43f78be2cf">

This read aligns to the beginning of the C gene but then continues with some DNA sequence from the genome between the J and C genes.

Although these reads align within the locus, they clearly do not represent a complete VDJ rearrangement. The origin is unclear and could be due to DNA contamination or a complicated, incomplete rearrangement.
